_sass/addon/commons.scss

```diff
@@ -959,6 +959,7 @@ $btn-mb: 0.5rem;
   color: var(--link-color);
   margin-left: 1rem;
   display: none;
+  white-space: nowrap;
```

Comment:
1. Reserve `4rem` space for the cancel button and revert the `white-space` here
```suggestion
  width: 4rem;
  display: none;
```

2. Add BS class for the button to handle the wrap issue:
https://github.com/cotes2020/jekyll-theme-chirpy/blob/74ab6f8adc2a7930e7bf21543ad17ded386c93c3/_includes/topbar.html#L68

```diff
-    <span id="search-cancel">{{ site.data.locales[include.lang].search.cancel }}</span>
+    <span id="search-cancel" class="text-truncate text-center">
+      {{- site.data.locales[include.lang].search.cancel -}}
+    </span>
```

Reply:
oh, you want overflow text ellipsis. But i think this code still has problem:

1. cancel text too short, too much space at the right side
<img width="438" alt="image" src="https://github.com/cotes2020/jekyll-theme-chirpy/assets/14135956/8a74111f-81cb-4a76-9ce8-ef06196f96a6">
2. cancel text too long, no space between cancel text and input
<img width="425" alt="image" src="https://github.com/cotes2020/jekyll-theme-chirpy/assets/14135956/52d9ed0a-b632-4ca3-a4f6-6af350ac97d4">